### DISCUSSION TITLE
Inputs needed for setting up server side rendering (SSR)

### DISCUSSION THREAD

**Discussion by otim** [react-navigation/react-navigation#12098] [General]
Dear react-navigation community!
Thank you for this exceptional navigation/routing framework. The documentation is spot on and so far I've found solutions to all cases I've ran into. Thank you!
Now, I am trying to get server side rendering to work with a react-native-web app that uses react-navigation. From the [documentation](https://reactnavigation.org/docs/server-rendering) I think going with koa is the way to go for my use case. But in my opinion the documentation is quite spotty here. I am missing quite a few important details I would need to set this up.
What configs are needed, what compiler has to be used, what command can be used for serving a local dev variant, how can it be built for production?
Can someone provide some pointers here? I'd be happy to improve the documentation once I manage to set this up.
Thank you!

**Comment by satya164** [comment]
Those details are quite specific to your project and its needs. React Navigation itself is just a library, and can be used any standard Node.js server setup that has some React Native specific adjustments.

I can't answer general questions regarding setting up development and production environments since those are quite broad questions. There are some things you may need to setup to handle React Native web apps specifically:

- Handle asset imports and transform them into URLs
- `.web.js` extensions that some libraries in React Native ecosystem use
- If there are libraries in your project or code that needs to be compiled, you may want to compile them with Babel and `@react-native/babel-preset` (or another lighter preset for your environment)

Bundling the server with a Bundler is one approach to handle this - which is done by frameworks like Next.js. Though handling platform specific extensions and assets can also be done by hooking into Node's module system using its APIs.

You may want to setup a server that handles server rendering React web apps first - you should be able to find resources for that. Once you have a working setup, you can try adding React Native specific adjustments on top of that. Though it's certainly not a trivial task.
